File: app/src/test/java/com/farmerbb/taskbar/helper/FreeformHackHelperTest.java
package com.farmerbb.taskbar.helper;

import org.junit.Test;
import org.junit.runner.RunWith;
import org.robolectric.RobolectricTestRunner;

import static org.junit.Assert.assertEquals;
import static org.junit.Assert.assertFalse;
import static org.junit.Assert.assertNotNull;
import static org.junit.Assert.assertTrue;

@RunWith(RobolectricTestRunner.class)
public class FreeformHackHelperTest {
    private FreeformHackHelper freeformHackHelper = FreeformHackHelper.getInstance();

    @Test
    public void testGetInstance() {
        assertNotNull(freeformHackHelper);
        for (int i = 1; i <= 20; i++) {
            assertEquals(freeformHackHelper, FreeformHackHelper.getInstance());
        }
    }

    @Test
    public void testSetFreeformHackActive() {
        assertFalse(freeformHackHelper.isFreeformHackActive());
        freeformHackHelper.setFreeformHackActive(true);
        assertTrue(freeformHackHelper.isFreeformHackActive());
        freeformHackHelper.setFreeformHackActive(false);
    }

    @Test
    public void testSetInFreeformWorkspace() {
        assertFalse(freeformHackHelper.isInFreeformWorkspace());
        freeformHackHelper.setInFreeformWorkspace(true);
        assertTrue(freeformHackHelper.isInFreeformWorkspace());
        freeformHackHelper.setInFreeformWorkspace(false);
    }

    @Test
    public void testSetTouchAbsorberActive() {
        assertFalse(freeformHackHelper.isTouchAbsorberActive());
        freeformHackHelper.setTouchAbsorberActive(true);
        assertTrue(freeformHackHelper.isTouchAbsorberActive());
        freeformHackHelper.setTouchAbsorberActive(false);
    }
}

File: app/src/test/java/com/farmerbb/taskbar/helper/FreeformHackHelperTest.kt
package com.farmerbb.taskbar.helper

import org.junit.Assert
import org.junit.Test
import org.junit.runner.RunWith
import org.robolectric.RobolectricTestRunner

@RunWith(RobolectricTestRunner::class)
class FreeformHackHelperTest {
    private val freeformHackHelper = FreeformHackHelper.getInstance()

    @Test
    fun testGetInstance() {
        Assert.assertNotNull(freeformHackHelper)
        for (i in 1..20) {
            Assert.assertEquals(freeformHackHelper, FreeformHackHelper.getInstance())
        }
    }

    @Test
    fun testSetFreeformHackActive() {
        Assert.assertFalse(freeformHackHelper.isFreeformHackActive)
        freeformHackHelper.isFreeformHackActive = true
        Assert.assertTrue(freeformHackHelper.isFreeformHackActive)
        freeformHackHelper.isFreeformHackActive = false
    }

    @Test
    fun testSetInFreeformWorkspace() {
        Assert.assertFalse(freeformHackHelper.isInFreeformWorkspace)
        freeformHackHelper.isInFreeformWorkspace = true
        Assert.assertTrue(freeformHackHelper.isInFreeformWorkspace)
        freeformHackHelper.isInFreeformWorkspace = false
    }

    @Test
    fun testSetTouchAbsorberActive() {
        Assert.assertFalse(freeformHackHelper.isTouchAbsorberActive)
        freeformHackHelper.isTouchAbsorberActive = true
        Assert.assertTrue(freeformHackHelper.isTouchAbsorberActive)
        freeformHackHelper.isTouchAbsorberActive = false
    }
}
